multicast/src/main/kotlin/com/dropbox/flow/multicast/Multicaster.kt

```diff
@@ -56,6 +56,11 @@ class Multicaster<T>(
      * it will receive values as well.
      */
     private val piggybackingDownstream: Boolean = false,
+    /**
+     * If true, an active upstream will stay alive even if all downstreams are closed. A downstream
+     * coming in later will receive a value from the live upstream.
+     */
+    private val keepUpstreamAlive: Boolean = false,
```

Comment:
can we give more granular control over this? (can be a followup CL)
at the least, i would like this to be a time instead of boolean so that it can timeout; can be very handy for usages in a ViewModel where you would want it to suspend for lets say a minute or so before cancelling it to avoid consuming resources (e.g. the real source might be more hot or keeping something else expensive)